common/mam/merkle.c

```diff
@@ -0,0 +1,195 @@
+#include "mam/merkle.h"
+#include "common/sign/v2/iss_curl.h"
+
+static trit_t const merkle_null_hash[HASH_LENGTH] = {0};
+
+// thanks to
+// https://stackoverflow.com/questions/3272424/compute-fast-log-base-2-ceiling#3272516
+static int ceil_log2(unsigned long long x) {
+  static const unsigned long long t[6] = {
+      0xFFFFFFFF00000000ull, 0x00000000FFFF0000ull, 0x000000000000FF00ull,
+      0x00000000000000F0ull, 0x000000000000000Cull, 0x0000000000000002ull};
+
+  int y = (((x & (x - 1)) == 0) ? 0 : 1);
+  int j = 32;
+  int depth_i;
+
+  for (depth_i = 0; depth_i < 6; depth_i++) {
+    int k = (((x & t[depth_i]) == 0) ? 0 : j);
+    y += k;
+    x >>= k;
+    j >>= 1;
+  }
+
+  return y;
+}
+
+static size_t binary_tree_size(size_t depth, size_t acc) {
+  if (depth == 0) {
+    return acc + 1;
+  }
+  return binary_tree_size(depth - 1, acc + (1 << depth));
+}
+
+static size_t merkle_node_base_index(size_t acc, size_t depth, size_t width,
```

Comment:
could you add some documentation?
is acc the size of the tree in every context?

Reply:
I think you are reviewing an outdated commit, this function does not exist anymore.